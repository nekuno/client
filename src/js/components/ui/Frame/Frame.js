import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Frame.scss';

export default class Frame extends Component {

    static propTypes = {
        onClickHandler : PropTypes.func
    };

    handleClick() {
        if (this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    render() {
        const {children} = this.props;

        return (
            <div className={styles.frame} onClick={this.handleClick.bind(this)}>
                {children}
            </div>
        );
    }
}