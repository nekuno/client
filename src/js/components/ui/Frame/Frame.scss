@import '../../../../scss/_partials/colors';

.frame {
  background: #FFF;
  color: #6C6F82;
  display: inline-block;
  font-size: 15px;
  margin: 2.5%;
  padding: 0;
  width: 95%;

  .title {
    background: #E2E3E7;
    color: #6C6F82;
    height: 10px;
    line-height: 10px;
    font-size: 12px;
    font-weight: bold;
    padding: 10px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
  .content {
    width: 100%;
  }
}