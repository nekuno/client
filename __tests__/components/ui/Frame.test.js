import React from 'react';
import Frame from '../../../src/js/components/ui/Frame/Frame.js';
import styles from '../../../src/js/components/ui/Frame/Frame.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Frame component', () => {
    const mockFn = jest.fn();
    const frame = shallow(<Frame onClickHandler={mockFn}>Lorem ipsum</Frame>);

    it('should be defined', () => {
        expect(Frame).toBeDefined();
    });
    it('should render correctly', () => {
        expect(frame).toMatchSnapshot();
    });
    it('Frame click event', () => {
        frame.find(`.${styles.frame}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
