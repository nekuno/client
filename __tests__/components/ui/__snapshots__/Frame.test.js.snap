// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Frame component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Frame
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </Frame>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="content"
        >
          Lorem ipsum
        </div>,
      ],
      "className": "frame",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum",
          "className": "content",
        },
        "ref": null,
        "rendered": "Lorem ipsum",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="content"
          >
            Lorem ipsum
          </div>,
        ],
        "className": "frame",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum",
            "className": "content",
          },
          "ref": null,
          "rendered": "Lorem ipsum",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Frame component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Frame
    onClickHandler={[MockFunction]}
    title="Lorem ipsum"
  >
    Lorem ipsum
  </Frame>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          Lorem ipsum
        </div>,
        <div
          className="content"
        >
          Lorem ipsum
        </div>,
      ],
      "className": "frame",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum",
          "className": "title",
        },
        "ref": null,
        "rendered": "Lorem ipsum",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum",
          "className": "content",
        },
        "ref": null,
        "rendered": "Lorem ipsum",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            Lorem ipsum
          </div>,
          <div
            className="content"
          >
            Lorem ipsum
          </div>,
        ],
        "className": "frame",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum",
            "className": "title",
          },
          "ref": null,
          "rendered": "Lorem ipsum",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum",
            "className": "content",
          },
          "ref": null,
          "rendered": "Lorem ipsum",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
